### ISSUE TITLE
General review and testing

### ISSUE THREAD

**Issue by crandmck** [loopbackio/loopback.io#55]
Before launch, it would be good to go through the content in http://loopback.io/doc/en/lb2/ and do  thorough review for accuracy, completeness, formatting, etc.
- Did anything from https://docs.strongloop.com/display/LB/LoopBack get left out?
- Did anything get messed up in migration?
- Any formatting or layout glitches?

**Comment by bajtos** [comment]
Nice to have: which pages should be changed in LB 3.x docs?

**Comment by crandmck** [comment]
@davidcheung Let me know if you have questions.  We're launching the site next week, so please have a look ASAP.

**Comment by davidcheung** [comment]
@crandmck will get on it today

**Comment by davidcheung** [comment]
# Suggestion
- [x] Table of content always contains the parent item "LoopBack 2.0" 
  i find it a little confusing, would rather have it removed or named `Home` if its customizable, for example if im trying to [create simple api](http://loopback.io/doc/en/lb2/Create-a-simple-API.html) and the first item on table of content sends me back to homepage.
- [x] `create an application`  should fall under getting started? not sure whats the best way of organizing the order but currently it goes as follow, where it teaches u to do all sorts of things, then create an application (but then it follows with managing users, authentication which are good flow)
  
  > - Getting started with LoopBack
  >   - Create a simple API
  >   - Use API Explorer
  >   - Connect your API to a data source
  >   - Extend your API
  >   - Add a static web page
  >   - Add a custom Express route
  >   - Next steps
  - Getting started part II
    - Introducing the Coffee Shop Reviews app
    - Create new data source
    - Create new models
    - Define model relations
    - Define access controls
    - Define a remote hook
    - Create AngularJS client
    - Learn more
  - Security advisories
  - Creating an application

# Fixables
- [x]  Sometimes when left nav expands it goes too far and is unscrollable [example](http://loopback.io/doc/en/lb2/Database-connectors.html) you(I) cant see all the options
- [x] in the table of overview, `Object` and `Array` link is wrong, should be (**PR #61 should fix this**)
  - `http://loopback.io/doc/en/lb2/LoopBack-types.html#array-types`
  - `http://loopback.io/doc/en/lb2/LoopBack-types.html#object-types`
- [x] http://loopback.io/doc/en/lb2/Connecting-to-MySQL.html enum link should be (**PR #61 should fix this**)
  `http://loopback.io/doc/en/lb2/MySQL-connector.html#enum`
  - same with corresponding type for http://loopback.io/doc/en/lb2/MySQL-connector.html#floating-point-types
  - same with corresponding type for http://loopback.io/doc/en/lb2/MySQL-connector.html#fixed-point-exact-value-types
- [x] Left nav `Built-in models REST API` , maybe due to 2 items same name, the first one always links to current URL. https://github.com/strongloop/loopback.io/blob/ef191bba2e7561192f684226d3eae221da1206df/_data/sidebars/lb2_sidebar.yml#L347 (solution might be to put url as `#`)
  - [] same with parent of http://loopback.io/doc/en/lb2/Database-connectors.html
- [ ] `common/models/account.json` and `discover-schema.js` link broken in http://loopback.io/doc/en/lb2/connecting-to-MySQL.html#create-the-collection-with-sample-data---automigration
  - same with http://loopback.io/doc/en/lb2/Connecting-to-MongoDB.html
  - probably unfixable atm because its auto created https://github.com/strongloop/loopback-example-database/tree/mysql
- [x]  `Glossary` falls under `Loopback-core-concept nav`, not sure if its intended
  http://loopback.io/doc/en/lb2/Glossary.html
  http://loopback.io/doc/en/lb2/LoopBack-core-concepts.html
- [x] Broken Link where it says `Process manager` in blue bubble, points to strong-pm, probably should be updated to apic
  http://loopback.io/doc/en/lb2/Create-a-simple-API.html#run-the-application
- [x] stray `</div>` tag. http://loopback.io/doc/en/lb2/Authentication-authorization-and-permissions.html#hiding-endpoints-for-related-models 
- [x] middle of http://loopback.io/doc/en/lb2/Accessing-related-models.html, and the code block right beneath, double underscores showing escape slashes
  
  > <pre>\_\_get\_\_projects</pre>
  > 
  > <pre>\_\_count\_\_projects</pre>
- [x] `**belongsTo**` double \* in the url and title http://loopback.io/doc/en/lb2/Relation-REST-API.html#listbelongstorelated-model-instances
- [x] broken link in http://loopback.io/doc/en/lb2/DB2-for-z-OS.html for `CRUD operations.`,  should be http://loopback.io/doc/en/lb2/Creating-updating-and-deleting-data.html
  - [x] also in http://loopback.io/doc/en/lb2/Informix.html
- [x] blue bubble closing issue http://loopback.io/doc/en/lb2/MySQL-connector.html the whole page is inside blue bubble
  - [x] same with http://loopback.io/doc/en/lb2/Oracle-connector.html

# Content missing
- [x] **Getting started with the Developer Toolkit command-line tool** not found anymore in new docs
  https://docs.strongloop.com/display/APIC/Getting+started+with+the+Developer+Toolkit+command-line+tool
  it is also not really related to lb2, maybe could be added to Page http://loopback.io/doc/en/lb2/Installing-IBM-API-Connect.html
- [x] ~~not sure if its needed but https://docs.strongloop.com/display/APIC/Defining+models is not in the new docs~~ the crossed out statement was untrue http://loopback.io/doc/en/lb2/Defining-models.html

**Comment by crandmck** [comment]
> Table of content always contains the parent item "LoopBack 2.0" 

That was just a glitch... fixed.

> Create an application should fall under getting started?

[http://loopback.io/doc/en/lb2/Create-a-simple-API.html](Create a simple API) basically covers that territory. (BTW, Issac wanted to call it "create an API" and not "create an app"). In contrast, [Creating an application](http://loopback.io/doc/en/lb2/Creating-an-application.html) is more of a general task-oriented article vs. a tutorial.

> Sometimes when left nav expands it goes too far

Created #60 to fix this.

> Glossary falls under Loopback-core-concept nav...

Yes, it used to be under "Reference" but it really got lost there.  I wanted to make it more prominent, and that was the best place I could find.

> Broken Link where it says Process manager in blue bubble,...

Fixed the link, and changed wording a bit.

> in the table of overview, Object and Array link is wrong

Which article?

>  http://loopback.io/doc/en/lb2/Connecting-to-MySQL.html enum link should be http://loopback.io/doc/en/lb2/MySQL-connector.html#enum

This comes from the MySQL connector README; need to fix it there. Same with Informix.

Re: the nav bar glitches, I'm still working those out.  I'll need to rework the navbar a bit, because Sequoia reimplemented the widget and it works a bit differently.  Will comment on those once I get 'er done.

**Comment by Sequoia** [comment]
> Left nav `Built-in models REST API` , maybe due to 2 items same name, the [first one](https://github.com/strongloop/loopback.io/blob/ef191bba2e7561192f684226d3eae221da1206df/_data/sidebars/lb2_sidebar.yml#L347) always links to current URL.

As @crandmck mentioned, we discussed the possibilities for side-nav "container"/folder items and the following seemed to be the options:
1. parent/"folder" **never** has a page, always redirects to first child
2. parent/"folder" **always** has a page
3. parent/"folder" **sometimes** has a page, sometimes doesn't.

Confluence allowed for option 3 above. I argue(d) that this behavior is confusing as it makes menu clicks less predictable (will this send me to a page? to the child? no page?). Option 1 and option 2 require more or less equal reworking of content organization: either move all "folder" pages to "folder> first child page" for solution 1, or ensure that all "folder" menu items have their own page they link to, overview of the section e.g.

I went with option 2 above, which will require [those "folder" items with no page](https://github.com/strongloop/loopback.io/blob/ef191bba2e7561192f684226d3eae221da1206df/_data/sidebars/lb2_sidebar.yml#L347) to have one created for them to point to. None of the solutions is perfect but that one seemed to create the most intuitive, predictable UX (click menu link, it takes you to a page, 100% of the time) so that's what we went with! It's certainly changeable but to my mind it's the sort of bike-shed issue that's best to just pick a solution & go with it. 😸

**Comment by crandmck** [comment]
Actually, on the old Confluence site https://docs.strongloop.com/ every "leaf" in the navigation "tree" corresponds to a page (which may or may not have children).  If you click on the `+` then it expands the children in the nav tree; if you click on the page title, it displays the page and expands the children in the nav tree.

This is also how it's currently working on loopback.io, which is great and exactly what I asked for.

However, there are still a few problems with how second-level containers work, that I suspect come from the server-side (Jekyll/Liquid) behavior.   And I would really like to be able to have one more level of nesting.... but I'll open a separate issue for these.

**Comment by davidcheung** [comment]
In this scenario to be specific, 
why dont we move the url from the `child` to the `sub-parent`, then delete the child since they are duplicates
https://github.com/strongloop/loopback.io/blob/ef191bba2e7561192f684226d3eae221da1206df/_data/sidebars/lb2_sidebar.yml#L345-L352

because right now on http://loopback.io/doc/en/lb2/Model-property-reference.html, when you click on `Built-in models REST API`'s text it does nothing

basically this https://github.com/strongloop/loopback.io/pull/64

**Comment by crandmck** [comment]
@davidcheung Yes, there are quite a few examples of similar problems. I'm working on the sidebar right now to address these.  Will let you know when I'm done so you can check it out.

**Comment by crandmck** [comment]
So, I've fixed the sidebar (David pointed out one issue I missed, and I fixed it).  So, I think we can close this, since we're essentially "launching" the site tomorrow with the blog post, etc..  Thanks for your help @davidcheung!
